tests/system/src/test/java/test/util/ScreenshotCapture.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.util;
+
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Base64;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.imageio.ImageIO;
+import javafx.application.Platform;
+import javafx.embed.swing.SwingFXUtils;
+import javafx.geometry.Rectangle2D;
+import javafx.scene.image.WritableImage;
+import javafx.scene.robot.Robot;
+import javafx.stage.Screen;
+
+/**
+ * A utility to capture a screenshot via JavaFX {@link Robot}.
+ * <p>
+ * Example:
+ * <pre>
+ * // write a base-64 encoded screenshot to stderr
+ * ScreenshotCapture.writeScreenshot(System.err);
+ * </pre>
+ */
+public class ScreenshotCapture {
+    /**
+     * Captures a screenshot using JavaFX {@link Robot} in the PNG format.
+     * <p>
+     * This method can be called from any thread.  If called from a thread other than
+     * the JavaFX Application Thread, the current thread will be paused until the screenshot is taken.
+     *
+     * @return the byte array containing the screenshot
+     * @throws IOException when an I/O error occurs
+     */
+    public static byte[] takeScreenshot() throws IOException {
```

Comment:
I have no idea how to interpret this byte array. What is the use case for this method?

Reply:
To save to a file, or send to a stream, for example.  This is a general purpose test utility.  The javadoc says "in PNG format".